drake/systems/trajectories/replanFunnels_mex.cpp

```diff
@@ -0,0 +1,1090 @@
+    snopt::integer Cold = 0;
+    
+    // Some silly stuff
+    snopt::integer nxname = 1, nFname = 1, npname = 0;
+    
+    // What to add to objective: 0 in our case
+    snopt::doublereal ObjAdd = 0.0;
+    
+    // Row in user function that corresponds to objective
+    snopt::integer ObjRow = 1;
+    
+    // Name of problem (used to print stuff)
+    char Prob[200]="";
+    
+    // iGfun, jGvar
+    snopt::integer lenG = 6;
+    
+    snopt::integer*    iGfun = new snopt::integer[lenG];
+    iGfun[0] = 1;
+    iGfun[1] = 1;
+    iGfun[2] = 1;
+    iGfun[3] = 2;
+    iGfun[4] = 2;
+    iGfun[5] = 2;
+    
+    snopt::integer*    jGvar = new snopt::integer[lenG];
+    jGvar[0] = 1;
+    jGvar[1] = 2;
+    jGvar[2] = 3;
+    jGvar[3] = 1;
+    jGvar[4] = 2;
+    jGvar[5] = 3;
+    
+    // Linear part of gradient (we don't have a linear part, so it's all zeros)
+    snopt::integer lenA = 0;
+    snopt::doublereal* A     = new snopt::doublereal[0];
+    snopt::integer* iAfun = new snopt::integer[0];
+    snopt::integer* jAvar = new snopt::integer[0];
+    
+    // xlow, xupp, Flow, Fupp
+    snopt::doublereal* xlow = new snopt::doublereal[nx];
+    xlow[0] = -1.0/0.0;
+    xlow[1] = -1.0/0.0;
+    xlow[2] = -1.0/0.0;
+    
+    snopt::doublereal* xupp = new snopt::doublereal[nx];
+    xupp[0] = 1.0/0.0;
+    xupp[1] = 1.0/0.0;
+    xupp[2] = 1.0/0.0;
+    
+    snopt::doublereal* Flow = new snopt::doublereal[nF];
+    Flow[0] = -1.0/0.0;
+    Flow[1] = -1.0/0.0;
+    
+    snopt::doublereal* Fupp = new snopt::doublereal[nF];
+    Fupp[0] = 1.0/0.0;
+    Fupp[1] = 1.0;
+    
+    // xnames, Fnames
+    snopt::integer INFO_snopt;
+    char xnames[8*1];  // should match nxname
+    char Fnames[8*1];  // should match nFname
+    
+    // Initial guess for x_shift is 0.
+    snopt::doublereal* x_guess    = new snopt::doublereal[nx];
+    x_guess[0] = 0.0; // (*dx_current);
+    x_guess[1] = 0.0; // (*(dx_current+1));
+    x_guess[2] = 0.0; // (*(dx_current+2));
+    
+    // xstate
+    snopt::integer *xstate = new snopt::integer[nx];
+    for(snopt::integer i = 0;i<nx;i++)
+    {
+        xstate[i] = 0;
+    }
+    
+    // xmul
+    snopt::doublereal *xmul = new snopt::doublereal[nx];
+    
+    // F, Fmul, Fstate
+    snopt::doublereal *F      = new snopt::doublereal[nF];
+    snopt::doublereal *Fmul   = new snopt::doublereal[nF];
+    snopt::integer    *Fstate = new snopt::integer[nF];
+    for(snopt::integer i = 0;i<nF;i++)
+    {
+        Fstate[i] = 0;
+    }
+    
+    // mincw, miniw, minrw
+    snopt::integer minrw,miniw,mincw;
+    
+    // nS, nInf, sInf
+    snopt::integer nS,nInf;
+    snopt::doublereal sInf;
+    
+    // lenrw, leniw, lencw
+    const int DEFAULT_LENRW = 500000;
+    const int DEFAULT_LENIW = 500000;
+    const int DEFAULT_LENCW = 500;
+    
+    snopt::integer lenrw = DEFAULT_LENRW, leniw = DEFAULT_LENIW, lencw = DEFAULT_LENCW;
+    
+    // cw, iw, rw
+    snopt::doublereal rw_static[DEFAULT_LENRW];
```

Comment:
this should use the updated snopt heap allocation version, as seen in NonlinearProgramSnoptcmex